src/components/Calculator.js

```diff
@@ -2,41 +2,133 @@
 /* eslint-disable no-useless-constructor */
 
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 import './calculator.css';
+import Button from './Button';
 
 class Calculator extends Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
+  }
+
+  onClickHandler = (e) => {
+    const { calcOperands, onButtonPress } = this.props;
+    const buttonName = e.target.innerText;
+    const calcObj = calculate(calcOperands, buttonName);
+    onButtonPress(calcObj);
   }
 
   render() {
+    const { calcOperands } = this.props;
     return (
       <div className="calc__container">
         <div className="calc__grid-container">
-          <div className="calc__grid-item calc__display">0</div>
-          <div className="calc__grid-item">AC</div>
-          <div className="calc__grid-item">+/-</div>
-          <div className="calc__grid-item">%</div>
-          <div className="calc__grid-item calc__operator">&divide;</div>
-          <div className="calc__grid-item">7</div>
-          <div className="calc__grid-item">8</div>
-          <div className="calc__grid-item">9</div>
-          <div className="calc__grid-item calc__operator">&times;</div>
-          <div className="calc__grid-item">4</div>
-          <div className="calc__grid-item">5</div>
-          <div className="calc__grid-item">6</div>
-          <div className="calc__grid-item calc__operator">&minus;</div>
-          <div className="calc__grid-item">1</div>
-          <div className="calc__grid-item">2</div>
-          <div className="calc__grid-item">3</div>
-          <div className="calc__grid-item calc__operator">+</div>
-          <div className="calc__grid-item calc__zero">0</div>
-          <div className="calc__grid-item">.</div>
-          <div className="calc__grid-item calc__operator">=</div>
+          <div className="calc__grid-item calc__display">{calcOperands.total ?? 0}</div>
```

Comment:
- [OPTIONAL] Try this one out.
```suggestion
          <div className="calc__grid-item calc__display">
            {`${
              (calcOperands.total || calcOperands.next || '0') +
              (calcOperands.operation || '') +
              ((calcOperands.operation && calcOperands.next) || '')
            }`}
          </div>
```

This suggestion is to make the app more interactive. Currently, when the user presses a key when starting to type an expression, the first value is not displayed on the Calculator's output.

Reply:
Hi @Lameck1 👋 
Thank you for your invaluable suggestion. I really do appreciate it 🙏